cpp/src/gandiva/compiled_expr.h

```diff
@@ -25,7 +25,7 @@
 
 namespace gandiva {
 
-using EvalFunc = int (*)(uint8_t** buffers, uint8_t** local_bitmaps,
+using EvalFunc = int (*)(uint8_t** buffers, int64_t* offsets, uint8_t** local_bitmaps,
```

Comment:
there is a single offset for the batch. why have an array of offsets ?

Reply:
Thanks for reviewing! I probably misunderstood how the offset works. I used to add a single offset, but I realized that each column also has its own offset, like such a case (use pseudocode here)

```python
a = [1, 2, 3, 4]
b = [2, 3, 4]
r = MakeRecordBatch([a.slice(1), b])
rr = r.slice(1)
```
will column `a` and `b` in record batch `rr` have different offsets?

Reply:
you are right, I didn't realise that the arrays can be sliced too. Can you please add a test for this case (similar to your example above) too ?

Reply:
No problem. Thanks again for reviewing!